source/slang/slang-parser.cpp

```diff
@@ -7779,13 +7786,95 @@ namespace Slang
         parser->sink->diagnose(token, Diagnostics::invalidCUDASMVersion);
         return nullptr;
     }
+
+    static NodeBase* parseVolatileModifier(Parser* parser, void* /*userData*/)
+    {
+        ModifierListBuilder listBuilder;
+
+        auto hlslMod = parser->astBuilder->create<HLSLVolatileModifier>();
+        hlslMod->keywordName = getName(parser, "volatile");
+        hlslMod->loc = parser->tokenReader.peekLoc();
+        listBuilder.add(hlslMod);
+
+        if (!parser->options.allowGLSLInput)
```

Comment:
There isn't much point checking for this option here, since by the time we reach this location, the token is already treated as if it is a modifier. If we are truly keen to make the keyword conditional, then we should not be registering the modifier parser function if we are not in glsl mode. But I think we don't need to worry that much at the moment.